app/helpers/lookup_keys_helper.rb

```diff
@@ -72,35 +72,41 @@ def overridable_lookup_keys(klass, host)
     klass.class_params.override.where(:environment_classes => {:environment_id => host.environment}) + klass.lookup_keys
   end
 
-  def hostgroup_key_with_diagnostic(hostgroup, key)
-    value, origin = hostgroup.inherited_lookup_value key
-    original_value = key.value_before_type_cast value
-    diagnostic_helper = popover('', _("<b>Description:</b> %{desc}<br><b>Type:</b> %{type}<br> <b>Matcher:</b> %{matcher}") % { :desc => key.description, :type => key.key_type, :matcher => origin}, :data => { :placement => 'top' })
-    parameter_value_content("value_#{key.key}", original_value, :popover => diagnostic_helper)
-  end
+  def key_with_diagnostic(obj, key, index, lookup_value, value_hash = nil)
+    if obj.class.model_name == "Hostgroup"
+      value, matcher = obj.inherited_lookup_value key
+    else # host
+      value_for_key = value_hash[key.id] && value_hash[key.id][key.key]
+      value, matcher = value_for_key ? [value_for_key[:value], "#{value_for_key[:element]} (#{value_for_key[:element_name]})"] : [key.default_value, _("Default value")]
+    end
 
-  def host_key_with_diagnostic(host, value_hash, key)
-    value_for_key = value_hash[key.id] && value_hash[key.id][key.key]
-    value, matcher = value_for_key ? [value_for_key[:value], "#{value_for_key[:element]} (#{value_for_key[:element_name]})"] : [key.default_value, _("Default value")]
     original_value = key.value_before_type_cast value
-    no_value = value.nil? && key.lookup_values.find_by_match("fqdn=#{host.fqdn}")
+    overriden_value = lookup_value.try(:value)
+    no_value = value.to_s.empty? && overriden_value.to_s.empty?
```

Comment:
Just realized why we did this - the value might be `false`, in which case `blank?` will return true and `to_s.empty?` will return false. reverting this back to the way it was.